cli/mender-artifact/copy_test.go

```diff
@@ -387,6 +387,54 @@ func TestCopy(t *testing.T) {
 				assert.Equal(t, "mender_test.img", inst[0].GetUpdateFiles()[0].Name)
 			},
 		},
+		{
+			name: "Delete a file from an image or Artifact",
+			argv: []string{"mender-artifact", "rm", ":/etc/mender/artifact_info"},
+			verifyTestFunc: func(imgpath string) {
+				os.Args = []string{"mender-artifact", "cat",
+					imgpath + ":/etc/mender/artifact_info"}
+				err := run()
+				assert.Error(t, err)
+			},
+		},
+		{
+			name: "Error when deleting a non-empty directory from an image or Artifact",
+			argv: []string{"mender-artifact", "rm", ":/etc/mender/"},
+			err: `debugfsRemoveDir: debugfs: error running command: "rmdir /etc/mender", err: 
+rmdir: directory not empty
+`,
+		},
+		{
+			name: "Delete a directory from an image or Artifact recursively",
+			argv: []string{"mender-artifact", "rm", "-r", ":/etc/mender/"},
+			verifyTestFunc: func(imgpath string) {
+				os.Args = []string{"mender-artifact", "cat",
+					imgpath + ":/etc/mender/artifact_info"}
+				err := run()
+				assert.Error(t, err)
+			},
+		},
+		{
+			name:        "Delete a file from a fat partition",
+			validimages: []int{2},
+			initfunc: func(imgpath string) {
+				require.Nil(t, ioutil.WriteFile("test.txt", []byte("foobar"), 0644))
+				os.Args = []string{"mender-artifact", "cp", "test.txt", imgpath + ":/boot/test.txt"}
+				require.NoError(t, run())
+			},
+			argv: []string{"mender-artifact", "rm", ":/boot/test.txt"},
+			verifyTestFunc: func(imgpath string) {
+				os.Args = []string{"mender-artifact", "cat",
+					imgpath + ":/boot/test.txt"}
+				err := run()
+				assert.Error(t, err)
+			},
+		},
+		{
+			name:        "Delete a directory from a fat partition",
+			validimages: []int{2}, // only valid for fat boot partition
+			argv:        []string{"mender-artifact", "rm", "-r", ":/uboot/testdir/"},
+		},
```

Comment:
There is no specific test for ext partition (i.e. `validimages: {1}`). I guess it is already covered in general rm tests (the ones without specifying `validimages`)?.